wordpress-mobile/WordPress-Android
# Issue/7139 signup magic link

### Fix
Add email signup screen as described in https://github.com/wordpress-mobile/WordPress-Android/issues/7139.  The new [`SignupMagicLinkFragment` class](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/7139-signup-magic-link/WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupMagicLinkFragment.java) is similar to the [`LoginMagicLinkSentFragment` class](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/7139-signup-magic-link/libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginMagicLinkSentFragment.java), but not entirely.  We can create a `BaseMagicLinkFragment` class to help abstract common code as an optimization when we move the signup code into the `WordPressLoginFlow` library.  I chose not to do that at this time to avoid conflicts with ongoing development.

### Test
0. Clear app data.
1. Tap ***Sign Up*** button.
2. Tap ***Sign Up with Email*** button.
3. Enter email address not associated with WordPress.com account.
4. Notice ***Signup Magic Link*** screen.
5. Check email inbox of address used in Step 3.
6. Notice signup email from WordPress.com.
7. Tap ***Try Again*** button.
8. Check email inbox of address used in Step 3.
9. Notice another signup email from WordPress.com.

#### Note
These changes do ***not*** include handling interception of the magic link from the signup email.  That will be covered in the subsequent epilogue issue.
